src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -217,35 +234,78 @@ def _adjust_batch_size(
 
     Args:
         trainer: instance of pytorch_lightning.Trainer
-
-        batch_arg_name: name of the field where batch_size is stored.
-
         factor: value which the old batch size is multiplied by to get the
             new batch size
-
         value: if a value is given, will override the batch size with this value.
             Note that the value of `factor` will not have an effect in this case
-
-        desc: either `succeeded` or `failed`. Used purely for logging
+        desc: either ``"succeeded"`` or ``"failed"``. Used purely for logging
 
     Returns:
         The new batch size for the next trial and a bool that signals whether the
         new value is different than the previous batch size.
     """
     model = trainer.lightning_module
     batch_size = lightning_getattr(model, batch_arg_name)
+    assert batch_size is not None
     new_size = value if value is not None else int(batch_size * factor)
     if desc:
-        log.info(f"Batch size {batch_size} {desc}, trying batch size {new_size}")
+        rank_zero_info(f"Batch size {batch_size} {desc}, trying batch size {new_size}")
+
+    # TODO improve this for multi eval dataloaders
+    if trainer.state.fn == "fit":
+        if trainer.train_dataloader is None:
+            trainer.reset_train_dataloader()
+
+        assert trainer.train_dataloader is not None
+        if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
+            new_size = min(new_size, len(trainer.train_dataloader.dataset))
+    else:
+        stage = trainer.state.stage
+        assert stage is not None
+        dataloaders = getattr(trainer, f"{stage.dataloader_prefix}_dataloaders")
+        if dataloaders is None:
+            getattr(trainer, f"reset_{stage.dataloader_prefix}_dataloader")()
 
-    if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
-        new_size = min(new_size, len(trainer.train_dataloader.dataset))
+        dataloaders = getattr(trainer, f"{stage.dataloader_prefix}_dataloaders")
+        assert dataloaders is not None
+        if not _is_valid_batch_size(new_size, dataloaders[0], trainer):
+            new_size = min(new_size, len(dataloaders[0].dataset))
 
     changed = new_size != batch_size
     lightning_setattr(model, batch_arg_name, new_size)
     return new_size, changed
 
 
-def _is_valid_batch_size(batch_size: int, dataloader: DataLoader, trainer: "pl.Trainer"):
+def _is_valid_batch_size(batch_size: int, dataloader: DataLoader, trainer: "pl.Trainer") -> bool:
+    from pytorch_lightning.utilities.data import has_len_all_ranks
+
     module = trainer.lightning_module or trainer.datamodule
     return not has_len_all_ranks(dataloader, trainer.strategy, module) or batch_size <= len(dataloader)
+
+
+def _collect_garbage(trainer: "pl.Trainer") -> None:
+    from pytorch_lightning.accelerators.gpu import GPUAccelerator
+
+    if isinstance(trainer.accelerator, GPUAccelerator):
+        garbage_collection_cuda()
```

Comment:
All this function does is basically call `torch.cuda.empty_cache()`, and it is no-op on non-cuda devices. So we don't need to have this check for GPUAccelerator